lib/run/index.js

```diff
@@ -85,7 +85,7 @@ module.exports = function (options, callback) {
             stopOnFailure: options.bail, // LOL, you just got trolled ¯\_(ツ)_/¯
             abortOnFailure: options.abortOnFailure, // used in integration tests, to be considered for a future release
             iterationCount: options.iterationCount || _.get(options, 'iterationData.length'),
-            environment: options.environment,
+            environment: _(options.environment.values).keyBy('key').mapValues('value').value(),
             globals: options.globals,
```

Comment:
We've missed globals here. It should arrive here just like environment and as such should go into runtime using same transformation that environment is going through.